get-thin-edge_io.sh

```diff
@@ -13,126 +32,306 @@ ARGUMENTS:
+        # Use older style packages names (with underscore)
+        TEDGE=tedge
+        TEDGE_MAPPER=tedge_mapper
+        TEDGE_AGENT=tedge_agent
+        TEDGE_WATCHDOG=tedge_watchdog
+        TEDGE_APT_PLUGIN=tedge_apt_plugin
+        TEDGE_APAMA_PLUGIN=tedge_apama_plugin
+        C8Y_CONFIGURATION_PLUGIN=c8y_configuration_plugin
+        C8Y_LOG_PLUGIN=c8y_log_plugin
     fi
 
-    echo "Installing for architecture $ARCH"
-else
-    echo "$ARCH is currently not supported. Currently supported are aarch64/arm64, armhf and amd64."
-    exit 0
-fi
+    echo "Thank you for trying thin-edge.io!"
+    echo
 
-if [ -d "/tmp/tedge" ]; then
-    rm -R /tmp/tedge
-fi
+    if [ -z "$VERSION" ]; then
+        VERSION="$DEFAULT_VERSION"
+
+        log "Version argument has not been provided, installing latest: $VERSION"
+        log "To install a particular version use this script with the version as an argument."
+        log "For example: sudo ./get-thin-edge_io.sh $VERSION"
+    fi
+
+    if [ "$ARCH" = "aarch64" ] || [ "$ARCH" = "arm64" ] || [ "$ARCH" = "armhf" ] || [ "$ARCH" = "amd64" ]; then
+        # Some OSes may read architecture type as `aarch64`, `aarch64` and `arm64` are the same architectures types.
+        if [ "$ARCH" = "aarch64" ]; then
+            ARCH='arm64'
+        fi
 
-echo "Installing mosquitto as prerequirement for thin-edge.io"
-apt install mosquitto -y
+        # For arm64, only the versions above 0.3.0 are available.
+        if [ "$ARCH" = "arm64" ] && ! dpkg --compare-versions "$VERSION" ge "0.3.0"; then
+            log "aarch64/arm64 compatible packages are only available for version 0.3.0 or above."
+            exit 1
+        fi
+
+        log "Installing for architecture $ARCH"
+    else
+        log "$ARCH is currently not supported. Currently supported are aarch64/arm64, armhf and amd64."
+        exit 0
+    fi
+
+    if ! command_exists mosquitto; then
+        log "Installing mosquitto as prerequisite for thin-edge.io"
+
+        # Lazily check apt-get dependency
+        if command_exists apt-get; then
+            export DEBIAN_FRONTEND=noninteractive
+            $sh_c "apt-get -y install mosquitto" >> "$LOGFILE"
+        elif command_exists apk; then
+            $sh_c "apk add mosquitto"
+        else
+            fail 1 "Missing prerequisite: 'apt-get' or 'apk'"
+        fi
+    fi
 
-case $TYPE in
-minimal) install_basic_components ;;
-full)
-    install_basic_components
-    install_tedge_agent
-    if apt -v >/dev/null 2>&1; then
+    case "$TYPE" in
+    minimal) install_basic_components ;;
+    full)
+        install_basic_components
+        install_tedge_agent
         install_tedge_plugins
+        ;;
+    *)
+        log "Unsupported argument type."
+        exit 1
+        ;;
+    esac
+
+
+    if is_dry_run; then
+        echo
+        echo "Dry run complete"
+    # Test if tedge command is there and working
+    elif tedge help >/dev/null; then
```

Comment:
Nice idea to double check that things are actually installed!

Reply:
actually this was already there in the existing script, so I can't take credit for that.